### ISSUE TITLE
Deprecate the s3 source connector and rename s3_v2 to s3

### ISSUE THREAD

**Issue by neverchanje** [risingwavelabs/risingwave#16234]
In the upcoming v1.9, we should deprecate the old `s3` connector. We can retain its code for backward compatibility in the short term, but renaming it to 's3_deprecate'. Additionally, we can remove its document to avoid misuse.

In the meanwhile, we should rename `s3_v2` to `s3`, making it the default implementation.

**Comment by tabVersion** [comment]
I agree with the purpose of deprecating legacy s3 connector but I don't think renaming it to an existing connector name is desirable. It may cause some trouble handing compatibility. 
How about just refuse the requests for `s3` connector and tell them use `s3_v2` instead.
